lib/kafo/scenario_manager.rb

```diff
@@ -0,0 +1,123 @@
+# encoding: UTF-8
+require 'kafo_wizards'
+
+module Kafo
+  class ScenarioManager
+    attr_reader :config_dir, :last_scenario_link, :previous_scenario
+
+    def initialize(config, last_scenario_link_name='last_scenario.yaml')
+      @config_dir = File.file?(config) ? File.dirname(config) : config
+      @last_scenario_link = File.join(config_dir, last_scenario_link_name)
+      @previous_scenario = File.realpath(last_scenario_link) if File.exists?(last_scenario_link)
+    end
+
+    def available_scenarios
+      # assume that *.yaml file in config_dir that has key :name is scenario definition
+      @available_scenarios ||= Dir.glob(File.join(config_dir, '*.yaml')).reject { |f| f =~ /#{last_scenario_link}$/ }.inject({}) do |scns, scn_file|
+        begin
+          content = YAML.load_file(scn_file)
+          scns[scn_file] = content if content.has_key?(:name)
+        rescue Psych::SyntaxError => e
+          warn "Warning: #{e}"
+        end
+        scns
+      end
+    end
+
+    def list_available_scenarios
+      say "Available scenarios"
+      available_scenarios.each do |config_file, content|
+        scenario = File.basename(config_file, '.yaml')
+        say "  #{content[:name]} (use: --scenario #{scenario})"
+        say "        " + content[:description] unless (content[:description].nil? || content[:description].empty?)
+      end
+      say "  No available scenarios found in #{config_dir}" if available_scenarios.empty?
+      KafoConfigure.exit(0)
+    end
+
+
+    def scenario_selection_wizard
+      wizard = KafoWizards.wizard(:cli, 'Select installation scenario',
+        :description => "Please select one of the pre-set installation scenarios. You can customize your installtion later during the installtion.")
+      f = wizard.factory
+      available_scenarios.keys.each do |scn|
+        wizard.entries << f.button(scn, :label => "#{available_scenarios[scn][:name]}: #{available_scenarios[scn][:description]}", :default => false)
```

Comment:
prints name with colon even if description is blank, should be just name in such case